script/acceptancetest

```diff
@@ -22,5 +22,33 @@ if [[ -z "${LOG_DIR}" ]]; then
 fi
 mkdir -p "${LOG_DIR}"
 
+# wait until $IRONIC_ENDPOINT is available, or 2 minutes. Use curl
+# to check if the endpoint is available
+IRONIC_ENDPOINT=${IRONIC_ENDPOINT}
+
+if [ -z "$IRONIC_ENDPOINT" ]; then
+  echo "IRONIC_ENDPOINT is not set. Skipping availability check."
+else
+  TIMEOUT=120 # 2 minutes in seconds
+  START_TIME=$(date +%s)
+
+  until
+    curl -s -o /dev/null -w "%{http_code}" "$IRONIC_ENDPOINT" | grep -q "20[0-9]"
+    if [ $? -eq 0 ]; then
+      echo "Ironic endpoint is available."
+      break
+    fi
+    CURRENT_TIME=$(date +%s)
+    ELAPSED_TIME=$((CURRENT_TIME - START_TIME))
+    if [ "$ELAPSED_TIME" -gt "$TIMEOUT" ]; then
+      echo "Timeout reached. Ironic endpoint is not available after 2 minutes."
+      break
+    fi
+    echo "Ironic endpoint not yet available. Waiting..."
+    sleep 5
+  done
+fi
```

Comment:
./script/acceptancetest: line 49: syntax error near unexpected token `done'

Reply:
I think you're missing a do? Add the ;do to the end of line. Plus unsure about until and the curl being on separate lines. Just generally the bash formatting looks a little weird here. Typically I'd expect this to look more like:

cmd="that long curl command"
until $cmd | grep -q '20[0-9]'; do
  [body]
done

until $cmd | grep -q '20[0-9]`
  do
    [body]
  done
is also OK but less common